src/main/java/net/imagej/ops/features/sets/DefaultGeometric2DComputerSet.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * #%L
+ * ImageJ software for multidimensional image processing and analysis.
+ * %%
+ * Copyright (C) 2014 - 2016 Board of Regents of the University of
+ * Wisconsin-Madison, University of Konstanz and Brian Northan.
+ * %%
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * 
+ * 1. Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ * 
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ * #L%
+ */
+package net.imagej.ops.features.sets;
+
+import net.imagej.ops.Op;
+import net.imagej.ops.Ops.Geometric.BoundarySize;
+import net.imagej.ops.Ops.Geometric.Boxivity;
+import net.imagej.ops.Ops.Geometric.Circularity;
+import net.imagej.ops.Ops.Geometric.Convexity;
+import net.imagej.ops.Ops.Geometric.Eccentricity;
+import net.imagej.ops.Ops.Geometric.FeretsAngle;
+import net.imagej.ops.Ops.Geometric.FeretsDiameter;
+import net.imagej.ops.Ops.Geometric.MainElongation;
+import net.imagej.ops.Ops.Geometric.MajorAxis;
+import net.imagej.ops.Ops.Geometric.MinorAxis;
+import net.imagej.ops.Ops.Geometric.Roundness;
+import net.imagej.ops.Ops.Geometric.Rugosity;
+import net.imagej.ops.Ops.Geometric.Size;
+import net.imagej.ops.Ops.Geometric.Solidity;
+import net.imagej.ops.features.sets.processors.ComputerSetProcessorUtils;
+import net.imglib2.roi.geometric.Polygon;
+import net.imglib2.type.numeric.real.DoubleType;
+
+import org.scijava.plugin.Plugin;
+
+/**
+ * Default implementation of {@link Geometric2DComputerSet}.
+ *
+ * @author Tim-Oliver Buchholz, University of Konstanz
+ *
+ */
+@Plugin(type = ComputerSet.class, label = "Geometric 2D Computers")
+public class DefaultGeometric2DComputerSet extends AbstractConfigurableComputerSet<Polygon, DoubleType>
+		implements Geometric2DComputerSet<Polygon, DoubleType> {
```

Comment:
> if we create a new Computer set like it is done here we don't pass an object.

The plan to get around that is to use pseudo-nulls, a.k.a. proxies or dummy placeholder objects. They will retain their generic type, and in some cases contain real state (e.g., a placeholder `Img` might know its `numDimensions()`). This will satisfy `Contingent` in various circumstances.

We can work on it at the hackathon.